### PR TITLE
Qa 791 fix menu item media

### PR DESCRIPTION
**Pull request by jfleschler**
The `MenuItemImpl` needed to be updated to the new way of attaching `Media`.
I have created a new `MenuItemMediaXref` class and made the necessary changes to `MenuItemImpl`.

When creating a new `MenuItem` for a `Menu`, the save would fail when selecting the "Link" `MenuItemType` and attaching an Image.

Steps to recreate:
Login to admin
Go to Content -> Menus -> Select "Header nav"
Click Add under MenuItems
Change Type to Link
Fill out all information and select an image.
Click Save button. The error appears.

QA Issue: https://github.com/BroadleafCommerce/QA/issues/791

### CODE REVIEW

File: src/main/java/org/broadleafcommerce/menu/domain/MenuItemImpl.java [hunk 1]
```diff
@@ -111,12 +119,30 @@
...
+    @AdminPresentationMapFields(
+            mapDisplayFields = {
+                    @AdminPresentationMapField(
+                            fieldName = "primary",
+                            fieldPresentation = @AdminPresentation(friendlyName = "MenuItemImpl_Image",
+                                    fieldType = SupportedFieldType.MEDIA,
+                                    order = Presentation.FieldOrder.IMAGE_URL)
+                    )
+            })
+    private Map<String, MenuItemMediaXref> menuItemMedia = new HashMap<String, MenuItemMediaXref>();
```
Comment (phillipuniverse) [inline src/main/java/org/broadleafcommerce/menu/domain/MenuItemImpl.java]: This should be protected

File: src/main/java/org/broadleafcommerce/menu/admin/server/handler/MenuItemCustomPersistenceHandler.java [hunk 2]
```diff
@@ -65,7 +66,8 @@ public Boolean canHandleInspect(PersistencePackage persistencePackage) {
         String ceilingEntityFullyQualifiedClassname = persistencePackage.getCeilingEntityFullyQualifiedClassname();
         try {
             Class testClass = Class.forName(ceilingEntityFullyQualifiedClassname);
-            return MenuItem.class.isAssignableFrom(testClass);
+            OperationType operationType = persistencePackage.getPersistencePerspective().getOperationTypes().getInspectType();
```
Comment (phillipuniverse) [inline src/main/java/org/broadleafcommerce/menu/admin/server/handler/MenuItemCustomPersistenceHandler.java:74]: Why are you only doing this for `BASIC` operations?

**Comment by jfleschler** [comment]
@phillipuniverse - I have made the requested changes to the code.

**Comment by domenicbenz** [comment]
hi @phillipuniverse 

Do you have any plans to merge this pull request? We are having the same issues when saving menu items of type LINK with an image set. :(

Or was this issue fixed in some other way and I just can't find it?

Thx in advance for your feedback!

Cheers

**Comment by kiteflo** [comment]
Dear @phillipuniverse

could you please merge this pull request? This one is essential to me as well...

please, please, please...!

thank you a lot
Florian

**Comment by nathandmoore** [comment]
@phillipuniverse Looks like @jfleschler wants to prevent the Menu Item CPH from handling maps. Hence, he only wants it to handle `Basic` ops. I will fix the merge conflict

**Comment by nathandmoore** [comment]
👍 tested

**Comment by jefffischer** [comment]
We should handle this issue in the 1.0.x stream without incurring a schema change.

**Comment by jefffischer** [comment]
This issue is addressed by: https://github.com/BroadleafCommerce/Menu/pull/23

**Pull request closed without merging**